iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/IcebergMajorQueryCompactor.java

```diff
@@ -63,20 +63,33 @@ public boolean run(CompactorContext context) throws IOException, HiveException,
 
     HiveConf conf = new HiveConf(context.getConf());
     String partSpec = context.getCompactionInfo().partName;
+    org.apache.hadoop.hive.ql.metadata.Table table = Hive.get(conf).getTable(context.getTable().getDbName(),
+        context.getTable().getTableName());
+    Table icebergTable = IcebergTableUtil.getTable(conf, table.getTTable());
     String compactionQuery;
 
     if (partSpec == null) {
-      HiveConf.setVar(conf, ConfVars.REWRITE_POLICY, RewritePolicy.ALL_PARTITIONS.name());
-      compactionQuery = String.format("insert overwrite table %s select * from %<s", compactTableName);
+      if (!IcebergTableUtil.isPartitioned(icebergTable)) {
+        HiveConf.setVar(conf, ConfVars.REWRITE_POLICY, RewritePolicy.ALL_PARTITIONS.name());
```

Comment:
if table is unpartitioned, why policy is RewritePolicy.ALL_PARTITIONS?

Reply:
For partitioned tables, starting with this PR, RewritePolicy.ALL_PARTITIONS won’t be used anymore, because if table is partitioned it will always create compaction request per partition and will use RewritePolicy.PARTITION for each partition.
RewritePolicy.ALL_PARTITIONS is used now in case the table is unpartitioned which means it compacts entire table.
Should we rename it to a proper name, like RewritePolicy.UNPARTITIONED or RewritePolicy.FULL_TABLE?